### PR TITLE
NEW: PricingTable component

### PR DESCRIPTION
**Pull request by filipdanisko**
Adding PricingTable component, due to new designs also extending List component.<br/><br/><br/><url>LiveURL: https://orbit-components-new-pricing-table.surge.sh</url>

### CODE REVIEW

**Comment by honzatmn** [comment]
Wow, that was really fast! 👏 

Is there any staging for it, so I can check designs? :)

**Comment by honzatmn** [comment]
Wow, this was fast! 👏

**Comment by filipdanisko** [comment]
> Wow, that was really fast! 👏
> 
> Is there any staging for it, so I can check designs? :)

If you would wait, like half a second more, you would see the link :D

**Comment by honzatmn** [comment]
> If you would wait, like half a second more, you would see the link :D

Well, you know me 😅

File: src/PricingTable/PricingTableItem/index.js.flow [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  +price?: Translation,
+  +mobileDescription?: Translation,
+  +priceBadge?: React$Node,
+  +featureIcon?: React$Node,
+  +badge?: React$Node,
+  +children?: React$Node,
+  +active?: boolean,
+  +compact?: boolean,
+  +action?: React$Node,
+  +basis?: number,
```
Comment (vepor) [inline src/PricingTable/PricingTableItem/index.js.flow]: I would use `string` for `basis`. It doesn't have to be `X%`, or does?

File: src/PricingTable/index.js.flow [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/*
+  DOCUMENTATION: https://orbit.kiwi/components/PricingTable/
+*/
+
+import typeof PricingTableItemType from "./PricingTableItem";
+import type { Globals } from "../common/common.js.flow";
+
+export type Props = {|
+  ...Globals,
+  +children: any, // TODO: Type this propperly
```
Comment (vepor) [inline src/PricingTable/index.js.flow]: `React$Node`

File: src/hooks/useMediaQuery.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  return (...args) => {
+    clearTimeout(interval);
+    interval = setTimeout(() => {
+      callback(...args);
+    }, time);
+  };
+};
+
+const useMediaQuery: UseMediaQuery = () => {
+  const theme = useTheme().orbit;
```
Comment (vepor) [inline src/hooks/useMediaQuery.js]: `const { orbit } = useTheme();`

File: src/hooks/useMediaQuery.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  const [breakpointList, setBreakpointList] = useState([
+    {
+      media: `(min-width: ${theme.widthBreakpointLargeDesktop}px)`,
```
Comment (vepor) [inline src/hooks/useMediaQuery.js]: You can use `getBreakpointWidth` function from `mediaQuery/index.js`

File: src/hooks/useMediaQuery.js.flow [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// @flow
+
+type Media = {|
+  +isLargeDesktop: null | number,
```
Comment (vepor) [inline src/hooks/useMediaQuery.js.flow]: You can use `maybe type`. `prop: ?number`

File: src/hooks/useMediaQuery.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      mapping: "isTablet",
+      matches: null,
+    },
+  ]);
+
+  // TODO: Check if window exists
+  const mediaList = breakpointList.map(breakpoint => window.matchMedia(breakpoint.media));
+
+  const findMatch = (mediaToMatch, list) => {
+    const res = list.filter(el => el.media === mediaToMatch);
```
Comment (vepor) [inline src/hooks/useMediaQuery.js]: Use `Array.find` prototype to return the first match

File: src/PricingTable/PricingTableItem/index.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+const StyledPricingTableItem = styled.div`
+  display: flex;
+  flex-grow: 1;
+  position: relative;
+  background: #ffffff;
+  box-shadow: 0 0 2px 0 rgba(37, 42, 49, 0.16), 0 1px 4px 0 rgba(37, 42, 49, 0.12); /* TODO: Add elevation token */
+  transition: ${({ theme }) => theme.orbit.durationNormal};
+  border-radius: ${({ theme }) => theme.orbit.borderRadiusNormal}; /* TODO: Add token */
+  padding: ${({ theme }) => theme.orbit.spaceMedium}; /* TODO: Add token */
+  background: #ffffff;
```
Comment (vepor) [inline src/PricingTable/PricingTableItem/index.js]: Redundant declaration of background, use token also 🙂

File: src/PricingTable/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import PricingTable, { PricingTableItem } from "@kiwicom/orbit-components/lib/PricingTable";
+```
+
+#### Props
+Table below contains all types of the props in the PricingTableItem component.
+
+| Name              | Type                   | Default     | Description                      |
+| :---------------- | :--------------------- | :---------- | :------------------------------- |
+| action            | `React$Node`           |             | Area for action elements, like Button.
+| badge             | `string | React$Node`  |             | Badge above the PricingTableItem, works with Orbit Badge
```
Comment (vepor) [inline src/PricingTable/README.md]: Escape `|`

File: src/PricingTable/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+| **children**      | `React.Node`           |             | Content of the PricingTableItem component.
+| dataTest          | `string`               |             | Optional prop for testing purposes.
+| featureIcon       | `React$Node`           |             | Feature Icon displayed at top of the PricingTableItem
+| mobileDescription | `Translation`          |             | Description of PricingTableItem, displayed on mobile
+| name              | `Translation`          |             | Name of PricingTableItem
+| onClick           | `() => void \| Promise`|             | Function for handling the onClick event.
+| price             | `Translation`          |             | Price of item
+| priceBadge        | `React$Node`           |             | Badge instead of `price`, works with Orbit Badge
```
Comment (vepor) [inline src/PricingTable/README.md]: can you link relatively to `../Badge`? Should work also on orbit.kiwi 🙂

File: src/PricingTable/PricingTableItem/index.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+            {price && (
+              <Text size="large" weight="bold" type="primary" align="center">
+                {price}
+              </Text>
+            )}
+            {priceBadge && <Stack justify="center"> {priceBadge} </Stack>}
+          </Stack>
+          {!compact && (
+            <Stack justify="between" direction="column">
+              {!compact && children}
```
Comment (vepor) [inline src/PricingTable/PricingTableItem/index.js]: You have already evaluated `!compact`, so it's probably unnecessary 🙂

File: src/PricingTable/PricingTableItem/index.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+              {!compact && action && (
+                <Stack justify="center" grow={false}>
+                  {action}
+                </Stack>
+              )}
+            </Stack>
+          )}
+          {compact && (
```
Comment (vepor) [inline src/PricingTable/PricingTableItem/index.js]: You can also use ternary `!compact ? first : second`

File: src/PricingTable/PricingTableItem/index.js (hunk 10 above)
Comment (filipdanisko) [reply src/PricingTable/PricingTableItem/index.js]: good catch!

**Pull request merged**